drv/stm32h7-sprot-server/src/main.rs

```diff
@@ -124,12 +129,24 @@ cfg_if::cfg_if! {
     }
 }
 
+/// Return an error if the expected MsgType doesn't match the actual MsgType
+macro_rules! expect_msg {
+    ($expected:expr, $actual:expr) => {
+        if $expected != $actual {
+            ringbuf_entry!(Trace::WrongMsgType($actual));
+            Err(SprotError::BadMessageType)
+        } else {
+            Ok(())
+        }
+    };
+}
+
```

Comment:
Not sure why this is a macro, e.g.
```suggestion
fn expect_msg(expected: MsgType, actual: MsgType) -> Result<(), SprotError> {
    if expected != actual {
        ringbuf_entry!(Trace::WrongMsgType(actual));
        Err(SprotError::BadMessageType)
    } else {
        Ok(())
    }
}
```

Reply:
It's a historical arfifact!

Reply:
Fixed in [3a0f61a](https://github.com/oxidecomputer/hubris/pull/965/commits/3a0f61ac0031aaed46524561bf56eb41f20f085f)